lib/ProxySQL_Admin.cpp

```diff
@@ -6958,6 +7222,104 @@ bool ProxySQL_Admin::set_variable(char *name, char *value) {  // this is the pub
 	return false;
 }
 
+void ProxySQL_Admin::p_update_metrics() {
+	// Update proxysql_uptime
+	auto t1 = monotonic_time();
+	auto new_uptime = (t1 - GloVars.global.start_time)/1000/1000;
+	auto cur_uptime = this->metrics.p_counter_array[p_admin_counter::uptime]->Value();
+	this->metrics.p_counter_array[p_admin_counter::uptime]->Increment(new_uptime - cur_uptime);
+
+	// Update memory metrics
+	this->p_stats___memory_metrics();
+	// Update stmt metrics
+	this->p_update_stmt_metrics();
+}
+
+void ProxySQL_Admin::p_stats___memory_metrics() {
```

Comment:
This function can become expensive, and we need a way to not execute it too often and (serve stale data).
We can add a new global variable (an admin variable) that defines a TTL for such function.
If the function is called within N seconds from the last time it processed the data, return immediately.

Reply:
Done!